File: Creational/Builder/PHP/builders/CarManualBuilder.php
<?php

require_once(__DIR__.'/../interfaces/IBuilder.php');
require_once(__DIR__.'/../interfaces/IEngine.php');
require_once(__DIR__.'/../products/CarManual.php');

/**
 * Builder for concrete product.
 */

class CarManualBuilder implements IBuilder
{
	public function reset()
	{
		$this->_manual = new CarManual();
	}

	/**
	 * @param string $engine
	 */
	public function setEngine(IEngine $engine)
	{
		$this->getCarManual()->setEngine($engine);
	}

	/**
	 * @param int $seats
	 */
	public function setSeats($seats)
	{
		$this->getCarManual()->setSeats($seats);
	}

	/**
	 * @param boolean $tComputer
	 */
	public function setTripComputer($tComputer = true)
	{
		$this->getCarManual()->setTripComputer($tComputer);
	}

	public function getResult(): CarManual
	{
		return $this->getCarManual();
	}

	private function getCarManual(): CarManual
	{
		return $this->_manual;
	}

	private $_manual;
}

File: Creational/Builder/C++/builders/CarManualBuilder.cpp
#include "CarManualBuilder.h"
#include "./../interfaces/IEngine.h"
#include "./../products/CarManual.h"

void CarManualBuilder::reset()
{
    _carManual = new CarManual();
}

void CarManualBuilder::setEngine(IEngine* engine)
{
    getCarManual()->setEngine(engine);
}

void CarManualBuilder::setSeats(const int seats)
{
    getCarManual()->setSeats(seats);
}

void CarManualBuilder::setTripComputer(const bool tComputer)
{
    getCarManual()->setTripComputer(tComputer);
}
